[ONNX] Update doc and error message for indexing export (#64290)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64581 [ONNX] Enable export of __xor_ (#64042)
* #64580 [ONNX] Fix inplace fill_ dtype export mismatch (#64233)
* **#64579 [ONNX] Update doc and error message for indexing export (#64290)**
* #64578 [ONNX] Fix remainder export (#64230)
* #64381 Add onnx test for batched_nms (#53175)
* #64379 [ONNX] Fix gather squeeze axis in constant folding (#63588)
* #64378 [ONNX] Fix cuda test case (#63597)
* #64377 [ONNX] Fix input sequence for pad op (#60554)
* #64376 Fix empty size constant creation (#63607)
* #64375 [ONNX] Update instance_norm implementation and support training (#60538)

Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D30919603](https://our.internmc.facebook.com/intern/diff/D30919603)

BowenBao committed:
[ONNX] Update doc and error message for indexing export (#64290)

Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update doc and error message for indexing export (#64290)"


Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919603).

BowenBao committed:
Update on "[ONNX] Update doc and error message for indexing export (#64290)"


Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D30919603](https://our.internmc.facebook.com/intern/diff/D30919603)

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919603).

BowenBao committed:
Update on "[ONNX] Update doc and error message for indexing export (#64290)"


Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D30919603](https://our.internmc.facebook.com/intern/diff/D30919603)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update doc and error message for indexing export (#64290)"


Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D30919603](https://our.internmc.facebook.com/intern/diff/D30919603)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919603).

BowenBao committed:
Update on "[ONNX] Update doc and error message for indexing export (#64290)"


Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D30919603](https://our.internmc.facebook.com/intern/diff/D30919603)

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update doc and error message for indexing export (#64290)"


Added suggested workarounds into indexing section of onnx export documentation.
Update indexing export warning message with link to documentation.

Co-authored-by: BowenBao <bowbao@microsoft.com>

Differential Revision: [D30919603](https://our.internmc.facebook.com/intern/diff/D30919603)

[ghstack-poisoned]
